wallet/internal/db/tx.go

```
@@ -0,0 +1,36 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+)
+
+// execInTx executes a function within a database transaction. It handles
+// the transaction lifecycle: begin, commit, and rollback on error.
+//
+// This is a helper function used by the public ExecuteTx methods on
+// PostgresWalletDB and SQLiteWalletDB. It guarantees that the transaction
+// will be either committed (on success) or rolled back (on error or panic).
+func execInTx(ctx context.Context, db *sql.DB, fn func(*sql.Tx) error) error {
+	tx, err := db.BeginTx(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("begin tx: %w", err)
+	}
+
+	defer func() {
+		_ = tx.Rollback()
```

nit: explain why it's ok to ignore the error here?

---

added the comment